Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -141,76 +147,80 @@ std::string DecodeString(pb_istream_t* stream) {
   return result;
 }
 
-}  // namespace
-
-using firebase::firestore::model::FieldValue;
-
-void Serializer::EncodeFieldValue(const FieldValue& field_value,
-                                  std::vector<uint8_t>* out_bytes) {
-  // TODO(rsgowman): how large should the output buffer be? Do some
-  // investigation to see if we can get nanopb to tell us how much space it's
-  // going to need.
-  uint8_t buf[1024];
-  pb_ostream_t stream = pb_ostream_from_buffer(buf, sizeof(buf));
+void EncodeObject(
+    pb_ostream_t* stream,
+    const std::map<const std::string, const FieldValue>& object_value);
+std::map<const std::string, const FieldValue> DecodeObject(
```

Comment:
If forward declarations really are required, please put them up at the top of the file so we don't have to hunt for them.

Also, laid out together like this it took me a moment to realize this wasn't a single declaration. Maybe newline in between?